api/src/tests/simulation_test.rs

```diff
@@ -153,3 +157,200 @@ async fn test_simulate_txn_with_aggregator() {
         unreachable!("Simulation uses Ed25519 authenticator.");
     }
 }
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+async fn test_bcs_simulate_simple() {
+    let transfer_amount: u64 = SMALL_TRANSFER_AMOUNT;
+
+    let mut context = new_test_context(current_function_name!());
+    let alice = &mut context.gen_account();
+    let bob = &mut context.gen_account();
+    let txn = context.mint_user_account(alice).await;
+    context.commit_block(&vec![txn]).await;
+
+    let txn = context.account_transfer_to(alice, bob.address(), transfer_amount);
+    let body = bcs::to_bytes(&txn).unwrap();
+
+    // expected to fail due to using a valid signature.
+    let _resp = context
+        .expect_status_code(400)
+        .post_bcs_txn("/transactions/simulate", body)
+        .await;
+
+    if let TransactionAuthenticator::Ed25519 {
+        public_key,
+        signature: _,
+    } = txn.authenticator_ref()
+    {
+        let txn = SignedTransaction::new_signed_transaction(
+            txn.clone().into_raw_transaction(),
+            TransactionAuthenticator::Ed25519 {
+                public_key: public_key.clone(),
+                signature: Ed25519Signature::dummy_signature(),
+            },
+        );
+
+        let body = bcs::to_bytes(&txn).unwrap();
+
+        // expected to succeed
+        let resp = context
+            .expect_status_code(200)
+            .post_bcs_txn("/transactions/simulate", body)
+            .await;
+
+        assert!(resp[0]["success"].as_bool().unwrap(), "{}", pretty(&resp));
+    } else {
+        unreachable!("Simulation uses Ed25519 authenticator.");
+    }
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+async fn test_bcs_simulate_without_auth_key_check() {
+    let transfer_amount: u64 = SMALL_TRANSFER_AMOUNT;
+
+    let mut context = new_test_context(current_function_name!());
+    let alice = &mut context.gen_account();
+    let bob = &mut context.gen_account();
+    let txn = context.mint_user_account(alice).await;
+    context.commit_block(&vec![txn]).await;
+
+    // Construct a signed transaction.
+    let txn = context.account_transfer_to(alice, bob.address(), transfer_amount);
+    // Replace the authenticator with a NoAccountAuthenticator in the transaction.
+    let txn = SignedTransaction::new_signed_transaction(
+        txn.clone().into_raw_transaction(),
+        TransactionAuthenticator::SingleSender {
+            sender: AccountAuthenticator::NoAccountAuthenticator,
```

Comment:
We need a test that proves that this variant is rejected for the normal txn submission path.

Reply:
Added the tests here:
* https://github.com/aptos-labs/aptos-core/pull/13714/files#diff-3f5e53d39326e69c98cf9e1210c60a8a8e54e2668ef8e128c4d295fbc82964bdR389
* https://github.com/aptos-labs/aptos-core/pull/13714/files#diff-3f5e53d39326e69c98cf9e1210c60a8a8e54e2668ef8e128c4d295fbc82964bdR389